src/Aspire.Hosting.Azure/AzureDeployingContext.cs

```diff
@@ -129,11 +132,22 @@ private async Task<bool> TryProvisionAzureBicepResources(List<AzureBicepResource
                                 {
                                     bicepResource.ProvisioningTaskCompletionSource = new(TaskCreationOptions.RunContinuationsAsynchronously);
 
-                                    await bicepProvisioner.GetOrCreateResourceAsync(bicepResource, provisioningContext, cancellationToken).ConfigureAwait(false);
+                                    if (await bicepProvisioner.ConfigureResourceAsync(configuration, bicepResource, cancellationToken).ConfigureAwait(false))
+                                    {
+                                        bicepResource.ProvisioningTaskCompletionSource?.TrySetResult();
+                                        await resourceTask.CompleteAsync($"Using existing deployment for {bicepResource.Name}", CompletionState.Completed, cancellationToken).ConfigureAwait(false);
+                                    }
+                                    else
+                                    {
+                                        await bicepProvisioner.GetOrCreateResourceAsync(bicepResource, provisioningContext, cancellationToken).ConfigureAwait(false);
+
+                                        // Save provisioning state after each successful resource deployment to user secrets
+                                        await userSecretsManager.SaveUserSecretsAsync(provisioningContext.UserSecrets, cancellationToken).ConfigureAwait(false);
```

Comment:
These calls can be concurrent, right? What happens when multiple threads get into `userSecretsManager.SaveUserSecretsAsync`?

Reply:
lol yes don't tell David I still fudged this up